patches/minecraft/net/minecraft/item/ItemDye.java.patch

```diff
@@ -18,26 +18,35 @@
                          p_180614_2_.func_180501_a(p_180614_3_, iblockstate1, 10);
  
                          if (!p_180614_1_.field_71075_bZ.field_75098_d)
-@@ -99,8 +99,18 @@
+@@ -99,8 +99,27 @@
  
      public static boolean func_179234_a(ItemStack p_179234_0_, World p_179234_1_, BlockPos p_179234_2_)
      {
 +        if (p_179234_1_ instanceof net.minecraft.world.WorldServer)
-+            return applyBonemeal(p_179234_0_, p_179234_1_, p_179234_2_, net.minecraftforge.common.util.FakePlayerFactory.getMinecraft((net.minecraft.world.WorldServer)p_179234_1_));
++            return applyBonemeal(p_179234_0_, p_179234_1_, p_179234_2_, net.minecraftforge.common.util.FakePlayerFactory.getMinecraft((net.minecraft.world.WorldServer)p_179234_1_), null);
 +        return false;
 +    }
 +
++    /**
++     * @deprecated Use {@link #applyBonemeal(ItemStack, World, BlockPos, EntityPlayer, EnumHand)} instead.
++     */
++    @Deprecated
 +    public static boolean applyBonemeal(ItemStack p_179234_0_, World p_179234_1_, BlockPos p_179234_2_, EntityPlayer player)
 +    {
++        return applyBonemeal(p_179234_0_, p_179234_1_, p_179234_2_, player, null);
```

Comment:
`EntityLivingBase#getActiveHand` only returns valid results when the player is actively using an item (e.g. drawing a bow, eating food, blocking with a shield).

It could default to a specific hand, but the item wouldn't actually be in that hand for fake players like the dispenser. I think `null` is appropriate in this case.